File: Problem-3_Delete_node_without_head.py
# APPROACH  
# Time Complexity : O(1)
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : N/A
# Any problem you faced while coding this : None
#
#
# Your code here along with comments explaining your approach
# 1. Here the pointer to the node to be deleted is given. and not the head of the list
# 2. So copy the next node's value to this node
# 3. And assign the next pointer apprpriately

def delete(node):
  node.val = node.next.val
  node.next = node.next.next

File: Problem3_DeleteNodewithoutHead.java
// Time Complexity : o(1)
// Space Complexity : o(1)
// Did this code successfully run on GeeksforGeeks : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
class Solution
{
    void deleteNode(Node del)
    {
         // Your code here
         
         del.data=del.next.data;
         del.next=del.next.next;
    }
}